src/protobuf/parse.rs

```diff
@@ -42,11 +46,49 @@ pub fn proto_to_value(
             }
         }
         prost_reflect::Value::Message(v) => {
+            if let Some(descriptor_pool) = descriptor_pool {
+                if let Some(type_url_cow) = v.get_field_by_name("type_url") {
+                    if let Some(value_cow) = v.get_field_by_name("value") {
+                        if let prost_reflect::Value::String(type_url) = &*type_url_cow {
+                            if let prost_reflect::Value::Bytes(value) = &*value_cow {
+                                let type_name = type_url.trim_start_matches("type.googleapis.com/");
+                                match get_message_descriptor_from_pool(descriptor_pool, type_name) {
```

Comment:
This is a lot of nesting, can you explain the motivation and the use case for this?